lib/client/keystore.go

```diff
@@ -378,33 +373,61 @@ type withKubeCerts struct {
 }
 
 // TODO(awly): unit test this.
-func (o withKubeCerts) getKey(dirPath, username string, key *Key) error {
-	kubeDir := filepath.Join(dirPath, username+kubeDirSuffix, o.teleportClusterName)
-	kubeFiles, err := ioutil.ReadDir(kubeDir)
-	if err != nil && !os.IsNotExist(err) {
-		return trace.Wrap(err)
-	}
-	if key.KubeTLSCerts == nil {
-		key.KubeTLSCerts = make(map[string][]byte)
-	}
-	for _, fi := range kubeFiles {
-		data, err := ioutil.ReadFile(filepath.Join(kubeDir, fi.Name()))
-		if err != nil {
+func (o withKubeCerts) getKey(store LocalKeyStore, idx keyIndex, key *Key) error {
+	switch s := store.(type) {
```

Comment:
leave a TODO to add a `getKubeTLSCerts/deleteKubeTLSCert/etc` methods to `LocalKeyStore` to avoid enumerating all known implementations here.